monorepo/monorepo-ws/apps/federation-gateway/src/services/server-manager.ts

```diff
@@ -1,52 +1,48 @@
 import { ApolloGateway } from "@apollo/gateway";
 import { ApolloServer } from "apollo-server";
-import { checkActualSuperGraph } from ".";
 import { interval, port } from "../configuration/config";
-import { SupergraphStorageManager } from "./interfaces";
 import { tryCreatePartOfSupergraph } from "./supergraph-services";
+import { SupergraphManager } from "@monorepo-ws/supergraph-manager"
+import { logger } from "libs/server/src/logger";
 
 
 export class ServerManager {
-    superGraphManager: SupergraphStorageManager
-    superGraphSchema: Buffer
     gateway: ApolloGateway
     server: ApolloServer
 
-    constructor(superGraphManager: SupergraphStorageManager){
-        this.superGraphManager = superGraphManager
-    }
-
-    private async checkSupergraph(){
-        const newFile = await this.superGraphManager.getSupergraph()
-        if(!checkActualSuperGraph(this.superGraphSchema, newFile)){
-            await this.restartServer(newFile)
-        }
+    constructor(private readonly supergraphManager: SupergraphManager){
     }
 
     async runApp() {
         await this.runServer()
-        setInterval(() => this.checkSupergraph, interval)
+        setInterval(this.runServer.bind(this), interval)
     }
 
     private async runServer() {
+        if(!this.server){
+            try {
+                const superGraphSchema: Buffer = await this.supergraphManager.getSupergraph()
+                return await this.startServer(superGraphSchema.toString())
+            } catch (error) {
+                const superGraphSchema: Buffer =  await tryCreatePartOfSupergraph() 
```

Comment:
❓tohle bych asi nedelal. Proste pokud server nenastartuje nepodari se tak to pusti ten interval znova

Reply:
Odstraněno